adapters/eea-direct.js

```diff
@@ -0,0 +1,265 @@
+    },
+    (done) => {
+      // try to combine locations with coordinates
+      // by matching stationId or a spatial join.
+      // then return successful combos
+      request.get({url: stationsLink}, (err, res, locations) => {
+        if (err) {
+          return done(null, []);
+        }
+        locations = JSON.parse(locations);
+        map(coordinates, (crds, cb) => {
+          if (err) {
+            return cb(null, crds);
+          }
+          let geocodedProps = locations.find((location) => {
+            // an object with city, region, bounds, and stationId
+            // or just a stationId if reverse geocoding was unsuccessful
+            if (location.stationId) {
+              return location.stationId === crds.stationId;
+            } else {
+              return location === crds.stationId;
+            }
+          });
+          // if new station, denoted by no matching id,
+          // try to spatial join
+          if (!(geocodedProps)) {
+            geocodedProps = locations.find((location) => {
+              if (location.stationId) {
+                const bbox = bboxPolygon(location.bounds);
+                const stationPoint = point(
+                  crds.longitude,
+                  crds.latitude
+                );
+                return inside(stationPoint, bbox);
+              }
+            });
+          }
+          if (geocodedProps) {
+            geocodedProps = typeof geocodedProps === 'string' ? crds : Object.assign(geocodedProps, crds);
+            cb(null, geocodedProps);
+          }
+        }, (err, updatedCoordinates) => {
+          if (err) {
+            return done(null, coordinates);
+          }
+          done(null, updatedCoordinates);
+        });
+      });
+    }
+  ], (err, mappedData) => {
+    if (err) {
+      return cb(err, []);
+    }
+    let measurements = mappedData[0];
+    const coordinates = mappedData[1];
+    // map coordinates and location names to measurements
+    map(measurements, (measurement, done) => {
+      let station = matchStation(coordinates, measurement['coordinates']);
+      measurement['location'] = (station.location ? station.location : 'unused');
+      measurement['city'] = (station.city ? station.city : 'unused');
+      done(null, measurement);
+    }, (err, finalMeasurements) => {
+      if (err) {
+        return cb(null, {name: 'unused', measurements: measurements});
+      }
+      cb(null, {name: 'unused', measurements: finalMeasurements});
+    });
+  });
+};
+
+const matchStation = (coordinates, measurementCoords) => {
+  let station;
+  if (coordinates.length > 0) {
+    station = coordinates.find((crds) => {
+      return crds.coordinates === measurementCoords;
+    });
+    return station;
+  }
+};
+
+const makeCoordinates = (coordinatesList, stationId) => {
+  return coordinatesList.find((coordinates) => {
+    return coordinates.stationId === stationId;
+  }).coordinates;
+};
+
+const makeDate = (date, timeZone) => {
```

Comment:
Something still looks off with the times. Just ran this for DE and got the following for O3. Notice that every measurement has the same time.

```json
2017-08-21T17:21:25.986Z - info: {"date":{"utc":"2017-08-19T23:00:00.000Z","local":"2017-08-20T00:00:00+01:00"},"parameter":"o3","location":"Halberstadt","value":26,"unit":"µg/m³","city":"Sachsen-Anhalt","attribution":[{"name":"EEA"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":51.896982,"longitude":11.057442},"country":"DE","sourceName":"EEA Germany","sourceType":"government","mobile":false}
2017-08-21T17:21:25.986Z - info: {"date":{"utc":"2017-08-19T23:00:00.000Z","local":"2017-08-20T00:00:00+01:00"},"parameter":"o3","location":"Halberstadt","value":31,"unit":"µg/m³","city":"Sachsen-Anhalt","attribution":[{"name":"EEA"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":51.896982,"longitude":11.057442},"country":"DE","sourceName":"EEA Germany","sourceType":"government","mobile":false}
2017-08-21T17:21:25.986Z - info: {"date":{"utc":"2017-08-19T23:00:00.000Z","local":"2017-08-20T00:00:00+01:00"},"parameter":"o3","location":"Halberstadt","value":35,"unit":"µg/m³","city":"Sachsen-Anhalt","attribution":[{"name":"EEA"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":51.896982,"longitude":11.057442},"country":"DE","sourceName":"EEA Germany","sourceType":"government","mobile":false}
2017-08-21T17:21:25.986Z - info: {"date":{"utc":"2017-08-19T23:00:00.000Z","local":"2017-08-20T00:00:00+01:00"},"parameter":"o3","location":"Halberstadt","value":41,"unit":"µg/m³","city":"Sachsen-Anhalt","attribution":[{"name":"EEA"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":51.896982,"longitude":11.057442},"country":"DE","sourceName":"EEA Germany","sourceType":"government","mobile":false}
2017-08-21T17:21:25.986Z - info: {"date":{"utc":"2017-08-19T23:00:00.000Z","local":"2017-08-20T00:00:00+01:00"},"parameter":"o3","location":"Halberstadt","value":56,"unit":"µg/m³","city":"Sachsen-Anhalt","attribution":[{"name":"EEA"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":51.896982,"longitude":11.057442},"country":"DE","sourceName":"EEA Germany","sourceType":"government","mobile":false}
2017-08-21T17:21:25.986Z - info: {"date":{"utc":"2017-08-19T23:00:00.000Z","local":"2017-08-20T00:00:00+01:00"},"parameter":"o3","location":"Halberstadt","value":61,"unit":"µg/m³","city":"Sachsen-Anhalt","attribution":[{"name":"EEA"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":51.896982,"longitude":11.057442},"country":"DE","sourceName":"EEA Germany","sourceType":"government","mobile":false}
2017-08-21T17:21:25.986Z - info: {"date":{"utc":"2017-08-19T23:00:00.000Z","local":"2017-08-20T00:00:00+01:00"},"parameter":"o3","location":"Halberstadt","value":63,"unit":"µg/m³","city":"Sachsen-Anhalt","attribution":[{"name":"EEA"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":51.896982,"longitude":11.057442},"country":"DE","sourceName":"EEA Germany","sourceType":"government","mobile":false}
2017-08-21T17:21:25.987Z - info: {"date":{"utc":"2017-08-19T23:00:00.000Z","local":"2017-08-20T00:00:00+01:00"},"parameter":"o3","location":"Halberstadt","value":66,"unit":"µg/m³","city":"Sachsen-Anhalt","attribution":[{"name":"EEA"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":51.896982,"longitude":11.057442},"country":"DE","sourceName":"EEA Germany","sourceType":"government","mobile":false}
```

I believe this means the data parsing is failing and you're just getting the midnight time. 

However, looking over this, we should be able to simplify. They're actually providing a time with a timezone, so that should have all we need I think? We can just parse `2017-08-20 12:00:00+01:00` as the local time and then also convert to UTC?